### PR TITLE
Add RandomState support for df sampling

### LINKED ISSUE #525: The DataFrame.sample function does not accept Generator objects for random_state
**Describe the bug**
When calling [DataFrame.sample](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.sample.html) with a random_state argument, a type warning is
raised if a numpy.random.RandomState object is used, claiming that only an integer
can be inputted as an argument.

**To Reproduce**

```python
import numpy as np
import pandas as pd

rng = np.random.default_rng(0)

df = pd.DataFrame(data={'col1': [1, 2], 'col2': [3, 4]})
df.sample(n=1, random_state=rng)
```

Error message:
```
Argument of type "Generator" cannot be assigned to parameter "random_state" of type "int | None" in function "sample"
  Type "Generator" cannot be assigned to type "int | None"
    "Generator" is incompatible with "int"
```

**Please complete the following information:**
 - OS: Manjaro Linux
 - python 3.10.9
 - version of type checker: pyright 1.1
 - version of installed `pandas-stubs`: 1.5.2

### PR DESCRIPTION
**Pull request by intermezzio**
- [ ] Closes #525
- [ ] Tests added: /core/frame.pyi L234

### CODE REVIEW

File: pandas-stubs/core/frame.pyi [hunk 1]
```diff
@@ -1916,7 +1917,7 @@ class DataFrame(NDFrame, OpsMixin):
         frac: float | None = ...,
         replace: _bool = ...,
         weights: _str | ListLike | None = ...,
-        random_state: int | None = ...,
+        random_state: RandomState = ...,
```
Comment (Dr-Irv) [inline pandas-stubs/core/frame.pyi:1920]: should be `RandomState | None`

File: pandas-stubs/_typing.pyi [hunk 2]
```diff
@@ -373,4 +373,13 @@ ValidationOptions: TypeAlias = Literal[
     "many_to_many",
     "m:m",
 ]
+
+RandomState = Union[
```
Comment (Dr-Irv) [inline pandas-stubs/_typing.pyi]: Should be `RandomState:  TypeAlias = Union[`

(Sorry I should have caught that before - it's a difference between pandas and the stubs)

**Review by Dr-Irv — CHANGES_REQUESTED** [review]
Can you also change the declarations of that parameter in `sample` in `core/series.pyi` and `core/groupby/generic.pyi` as well?  You don't need to add tests for those.

You need to run `poetry run poe style` on your code as well.

File: pandas-stubs/_typing.pyi (hunk 2 above)
Comment (intermezzio) [reply pandas-stubs/_typing.pyi]: Done!

**Comment by intermezzio** [comment]
I've added this to generic and series. Unfortunately I'm having issues running the testing suite (poetry install runs but the precommit hook tries to get python from an unrelated conda environment which is messing up `poetry run poe style`) although I've at least changed the styling so that it should fix the long line created in the first commit of the PR.

**Review by Dr-Irv — APPROVED** [review]
thanks @intermezzio

**Pull request merged**